onchain/src/tests/username_store.cairo

```diff
@@ -14,23 +16,39 @@ fn deploy_contract() -> ContractAddress {
 fn test_claim_username() {
     let contract_address = deploy_contract();
     let dispatcher = IUsernameStoreDispatcher { contract_address };
-    dispatcher.claim_username('deal');
 
-    let username_address = dispatcher.get_username('deal');
+    snf::start_prank(CheatTarget::One(contract_address), utils::PLAYER1());
+    dispatcher.claim_username('deal');
 
-    assert(contract_address != username_address, 'Username not claimed');
+    let username_address = dispatcher.get_username_address('deal');
+    assert!(username_address == utils::PLAYER1(), "User didn't claim name");
+    let username = dispatcher.get_username(utils::PLAYER1());
+    assert!(username == 'deal', "Username not claimed");
+    snf::stop_prank(CheatTarget::One(contract_address));
 }
+
 #[test]
-fn test_transfer_username() {
+fn test_change_username() {
     let contract_address = deploy_contract();
     let dispatcher = IUsernameStoreDispatcher { contract_address };
-    dispatcher.claim_username('devsweet');
 
-    let second_contract_address = contract_address_const::<1>();
+    snf::start_prank(CheatTarget::One(contract_address), utils::PLAYER1());
+
+    // Set initial username
+    let username = 'devsweet';
+    dispatcher.claim_username(username);
+    let initial_username = dispatcher.get_username(utils::PLAYER1());
+    assert!(initial_username == username, "Initial username not set");
 
-    dispatcher.transfer_username('devsweet', second_contract_address);
+    // Change username
+    let new_username = 'devcool';
+    dispatcher.change_username(new_username);
 
-    let username_address = dispatcher.get_username('devsweet');
+    let new_username_address = dispatcher.get_username_address(new_username);
+    assert!(new_username_address == utils::PLAYER1(), "New username not linked");
+    let new_username = dispatcher.get_username(utils::PLAYER1());
+    assert!(new_username == new_username, "New username not set");
 
-    assert(username_address == second_contract_address, 'Username not Transferred');
+    let old_username_address = dispatcher.get_username_address(initial_username);
+    assert!(old_username_address == contract_address_const::<0>(), "Old username not unlinked");
```

Comment:
Can you add a couple tests that check that a user cannot claim 2 usernames, and check that the user cannot change usernames when they dont have one ( use the `should_panic` option )

Reply:
No problem, I am on it

Reply:
Sorry, I meant to add 2 new tests not change the existing test.
So there should be 4 tests : `test_claim_username`, `test_change_username` and the new tests `test_cannot_claim_multiple_usernames` and `test_cannot_change_with_no_username`. That way we are testing all the different possible usages and possible problems.